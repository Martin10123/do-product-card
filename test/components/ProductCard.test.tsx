import React from 'react'
import renderer from "react-test-renderer"
import { ProductCard } from '../../src/components';
import { product2 } from '../data/products';

const { act } = renderer;

describe('Pruebas en el componente ProductCard', () => {
    it('Debe de mostrar el componente correctamente', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {() => (
                    <h1>Product Card</h1>
                )}
            </ProductCard>
        )

        expect(wrapper.toJSON()).toMatchSnapshot()

    })

    it('Debe de incrementar', () => {
        const wrapper = renderer.create(
            <ProductCard product={product2}>
                {({ count, increaseBy }) => (
                    <>
                        <h1>Product Card</h1>
                        <p>{count}</p>
                        <button onClick={() => increaseBy(1)}>+1</button>
                    </>
                )}
            </ProductCard>
        );

        let tree = wrapper.toJSON();

        expect(tree).toMatchSnapshot();

        act(() => {
            (tree as any).children[2].props.onClick();
        })

        tree = wrapper.toJSON();

        expect((tree as any).children[1].children[0]).toBe("1")

    })
})