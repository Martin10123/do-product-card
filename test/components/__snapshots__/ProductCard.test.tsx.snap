// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ProductCard Debe de incrementar 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;

exports[`Pruebas en el componente ProductCard Debe de mostrar el componente correctamente 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product Card
  </h1>
</div>
`;
